### PR TITLE
#008 - Connect to JSON data API

### PR DESCRIPTION
**Pull request by dominik-git**
Test strategy:
-When I arrived to home page you should see "Loading" at the top of screen and after fetch data from server you should see "title of the shelve"

### CODE REVIEW

File: src/components/homeContainer/homeContainer.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// @flow
+import React from 'react';
+import Home from '../home/home';
```
Comment (varholak-peter) [inline src/components/homeContainer/homeContainer.js]: maybe we want to have a different hierarchy? Home won't be rendered without a home container so those can be inside the same folder where HomeContainer is main and Home is inside parts folder.

File: src/components/homeContainer/homeContainer.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class HomeContainer extends React.Component {
+
+  constructor() {
+    super();
+    this.state = {
+      data: {},
+    };
+  }
+
+  state: {
```
Comment (varholak-peter) [inline src/components/homeContainer/homeContainer.js]: this should be above constructor

File: src/components/homeContainer/homeContainer.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  componentWillMount() {
+    fetch('https://economist.twivel.io/api/v1/root/json', {
+      method: 'GET',
+    })
+    .then(response => response.json())
+    .then((data) => {
+      this.setState({ data });
+    });
+  }
+  render() {
```
Comment (varholak-peter) [inline src/components/homeContainer/homeContainer.js]: either have spaces between functions or don't, not both.

File: src/components/homeContainer/homeContainer.js (hunk 2 above)
Comment (dominik-git) [inline src/components/homeContainer/homeContainer.js]: It is not allowed by Linter

File: src/components/homeContainer/homeContainer.js (hunk 3 above)
Comment (dominik-git) [inline src/components/homeContainer/homeContainer.js]: fixed

File: src/components/homeContainer/homeContainer.js (hunk 1 above)
Comment (dominik-git) [inline src/components/homeContainer/homeContainer.js]: fixed

File: README.md [hunk 4]
```diff
@@ -138,3 +145,63 @@ You can also use [`jest.fn()` and `expect(fn).toBeCalled()`](http://facebook.git
...
+})
+
+### Explanation
+
+As you could see in example above, firstly in index.js you need to import react router and then components which you would like to to show. You need to wrap those components into Router and then you need to define url path and component which should be displayed. Navigate to component where you would like to have same navigation. Use "Link" for define url path. Link is equivalent to <a></a> element. [Read more about routing](https://github.com/reactjs/react-router-tutorial)
+
+
+## Setup
+
+Use "yarn add whatwg-fetch"
```
Comment (dominik-git) [inline README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -138,3 +145,63 @@ You can also use [`jest.fn()` and `expect(fn).toBeCalled()`](http://facebook.git
...
+
+### Example
+
+fetch(url).then(function(response) {
```
Comment (dominik-git) [inline README.md]: fixed

File: README.md [hunk 6]
```diff
@@ -138,3 +143,58 @@ You can also use [`jest.fn()` and `expect(fn).toBeCalled()`](http://facebook.git
...
+Open index.js and
+  1.import Router, Route and browserHistory, we will use browserHistory because we will not use # in url path.
+  2.import components which you would like to displayed
+  3.wrap components into route element
+  4.set url path and component which should be displayed
+  5.use Link for set url path for navigation between components defined in step 4.
+
+### Example  
+// index.js
+``` js
```
Comment (varholak-peter) [inline README.md]: remove space

**Pull request merged**